### ISSUE TITLE
Can't handle Node.js files

### ISSUE THREAD

**Issue by Sendao** [alibaba/structure-view#32]
The javascript parser does not support node.js files for some reason.

So, I recommend that tag-generator be allowed to fall back to CTAGS support when no tags are in the file. It's not great (flat; no regenerating tags on save) but it's something.

For my version I have adjusted the setting like so:

tag-generator.js:76::add:
    if( !tags.list || Object.keys(tags.list).length <= 0 ) {
      Gen = require('./tag-generators/universal');
      //ctx.lang = 'universal';
      tags = await Gen.parseFile(ctx); // tags contains list and tree data structure
      // For inline script in HTML
      if (tags.scriptNode && tags.scriptNode.length) await this.inlineScriptHandler(tags.scriptNode, ctx);
    }

**Comment by Joouis** [comment]
@Sendao Hi, I tried to reproduce this issue with [rimraf](https://github.com/isaacs/rimraf/blob/master/rimraf.js) and cannot make it, can you share a snippet which can work?

![image](https://user-images.githubusercontent.com/8896124/49167194-dbd6b180-f36f-11e8-923d-ec2a1d2caebb.png)

**Comment by Sendao** [comment]
I'm using Atom 1.33.0 x64, latest.
I use: atom-beautify, atom-clock, autoclose-html, ctrl-dir-scroll, minimap, minimap-bookmarks, minimap-selection, and structure-view. 
I disabled: archive-view, autocomplete-snippets, autoflow, dalek, deprecation-cop, encoding-selector, exception-reporting, git-diff, github, grammar-selector, language-clojure, language-csharp, language-{git,go,java,less,mustache,objective-c,perl,php,ruby,ruby-on-rails,sass,shellscript,toml,typescript,yaml}, line-ending-selector, metrics, open-on-github, snippets, spell-check, styleguide, timecop, and of course welcome. Can't have the welcome page without the cops and snippers. Anyway ;)

I listed them all but have disabled them after the fact. But the fact is it doesn't work at all for me. What's YOUR setup?

![fellback](https://user-images.githubusercontent.com/607861/49312466-256eeb80-f499-11e8-98e9-e138295a52e6.png)
See how it says "Fell back to using tags" on the right? I added that console line to replace the "no tags in file" indicator that always shows up on all javascript files for me.

PS I now use another tool, one for writers, which I have gifted the knowledge of caching (they didn't accept it) and made a parser for ctags for. It uses indentation to determine the structural depth level, but hey, it **works**! I can't be dealing with this broken stuff when I'm a progarmmer myself.
